app/views/records/show_fields/_location_of_resources.html.erb

```diff
@@ -1,12 +1,17 @@
 <% @generic_file.location_of_resources.each do |tinfo| %>
   <span itemprop="location_of_resources" itemscope itemtype="http://schema.org/Place">
-    <% if !tinfo.url.nil? %>
+    <% if not @presenter.display_terms[:location_of_resources][:subfields].nil? %>
+    <% @subfields = @presenter.display_terms[:location_of_resources][:subfields].keys %>
+    <% else %>
+    <% @subfields = [:location_physical, :location_Shelf_locator, :url] %>
+    <% end %>
+    <% if !tinfo.url.nil? and @subfields.include?(:url)%>
     <span itemprop="name"><%= tinfo.url %></span>
     <% end %>
-    <% if !tinfo.location_physical.nil? %>
+    <% if !tinfo.location_physical.nil? and @subfields.include?(:location_physical)%>
     <span itemprop="name"><%= tinfo.location_physical %></span>
     <% end %>
-    <% if !tinfo.location_shelf_locator.nil? %>
+    <% if !tinfo.location_shelf_locator.nil? and @subfields.include?(:location_shelf_locator)%>
     <span itemprop="name"><%= tinfo.location_shelf_locator %></span>
     <% end %>
   </span>
```

Comment:
Instead of this bock of if statements (line 8-16), you could have a loop over all subfields and then call 
```
tinfo.send [the current subfield]
```
check if this returns nil and if not display it. Does this make sense?